src/Conversion/ONNXToTorch/NN/CommonUtils.cpp

```diff
@@ -1,62 +1,44 @@
 #include "CommonUtils.h"
 
-typedef struct dim_pads {
-  int dim_start;
-  int dim_end;
-} dim_pads;
-
 std::vector<Value>
 createPadsArrayAttribute(::mlir::ArrayAttr pads, Type ty, Location loc,
                          ConversionPatternRewriter &rewriter) {
-  // Reading the ONNX side pads values and store in the array.
+  // Read ONNX side pads values and store inside a vector
   std::vector<Value> translatepadsList;
-  if (!pads)
-    return translatepadsList;
-
-  bool is_symmetric = true;
-  for (unsigned int i = 0; i < pads.size(); i += 2) {
-    if (pads[i] != pads[i + 1]) {
-      is_symmetric = false;
-      break;
-    }
-  }
-
-  dim_pads dimArray[pads.size()];
-  if (is_symmetric) {
-    for (unsigned int i = 0; i < pads.size(); i += 2) {
-      auto pad_value =
-          (pads[i].dyn_cast<IntegerAttr>()).getValue().getZExtValue();
-      auto f0 = IntegerAttr::get(ty, pad_value);
-      Value p0v = rewriter.create<ConstantIntOp>(loc, f0);
-      translatepadsList.push_back(p0v);
+  if (!pads)  {
+    for (unsigned i = 0; i < 2; i++) {
```

Comment:
There is a lot of hard-coding of 2D padding in this function, and I don't see any protection that this is expected. I wonder if the number of spatial dimensions should be passed as an argument?